client/c/src/yodaos_sdk.c

```diff
@@ -15,7 +15,10 @@
 static pthread_t mainThreadId;
 
 static void timeoutCb() {
-  RKLogw("Begin to ping active From pid(%lu)\n", pthread_self());
+  RKLogw("Begin to ping active From pid(%d) tpid(%lu)\n", getpid(),
+         pthread_self());
+  yodaos_local_event_lock(1);
```

Comment:
Why lock with nothing?

Reply:
Locking just waits for user thread finish, then send alive msg to runtime.